### PR TITLE
feat: Add products pagination end point ⚡

### PR DESCRIPTION
**Pull request by MarwaEbrahem**

### CODE REVIEW

**Review by claramounir — APPROVED** [review]

File: Networking/Sources/Networking/Models/ProductsPagination.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+// MARK: - Rating
+
+struct Rating: Codable {
+    let rate: Double
+    let count: Int
+}
+
+public typealias ProductsPagination = [ProductsPaginationElement]
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Models/ProductsPagination.swift]:
```suggestion
public typealias ProductsList = [Product]
```

File: Networking/Sources/Networking/Models/ProductsPagination.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    let productsPaginationDescription, category: String
+    let image: String
+    let rating: Rating
+    
+    enum CodingKeys: String, CodingKey {
+        case id, title, price
+        case productsPaginationDescription = "description"
+        case category, image, rating
+    }
+}
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Models/ProductsPagination.swift]:
```suggestion
public struct Product: Decodable {
    let id: Int
    let title: String
    let price: Double
    let description, category: String
    let image: String
    let rating: Rating
    
    enum CodingKeys: String, CodingKey {
        case id, title, price
        case description
        case category, image, rating
    }
}
```

File: Networking/Sources/Networking/Models/ProductsPagination.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+// MARK: - Rating
+
+struct Rating: Codable {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Models/ProductsPagination.swift]:
```suggestion
struct Rating: Decodable {
```

File: Networking/Sources/Networking/Remote/ProductsPaginationRemote.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by marwa on 09/08/2022.
+//
+
+import Foundation
+
+/// Protocol for `ProductsPaginationRemote` mainly used for mocking.
+///
+public protocol ProductsPaginationRemoteProtocol {
+    func loadProductsPagination(completion: @escaping (Result<ProductsPagination, Error>) -> Void)
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Remote/ProductsPaginationRemote.swift]:
```suggestion
    func loadProducts(completion: @escaping (Result<[Product], Error>) -> Void)
```

File: Networking/Sources/Networking/Remote/ProductsPaginationRemote.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+/// ProductsPagination: Remote Endpoints
+///
+public class ProductsPaginationRemote: Remote, ProductsPaginationRemoteProtocol {
+    
+    /// Loads  products pagination
+    ///
+    public func loadProductsPagination(completion: @escaping (Result<ProductsPagination, Error>) -> Void) {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Remote/ProductsPaginationRemote.swift]:
```suggestion
    public func loadProducts(completion: @escaping (Result<[Product], Error>) -> Void) {
```

**Review by ahmdmhasn — COMMENTED** [review]
Great effort 👏

**Review by oradwanomar — APPROVED** [review]

**Pull request merged**